File: csharp/isogram/Isogram.cs
using System;
using System.Collections;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        var letters = new ArrayList();
        foreach (var c in word.ToLower())
        {
            if (letters.Contains(c))
            {
                return false;
            }

            if (c.Equals('-') | c.Equals(' '))
            {
                continue;
            }

            letters.Add(c);
        }
        return true;
    }
}

File: go/isogram/isogram.go
package isogram

import (
	"strings"
	"unicode"
)

// IsIsogram checks if string is an isogram
func IsIsogram(word string) bool {
	charexists := map[rune]bool{}
	for _, char := range strings.ToLower(word) {
		if unicode.IsLetter(char) && charexists[char] {
			return false
		}
		charexists[char] = true
	}
	return true
}